src/pages/payroll/PayrollPage.js

```diff
@@ -147,13 +157,17 @@ function PayrollPage({
 }
 
 const mapDispatchToProps = (dispatch) => bindActionCreators({
+  fetchPayroll,
+  createPayroll,
+  clearPayroll,
+  deletePayrolls,
   coreConfirm,
   clearConfirm,
   journalize,
 }, dispatch);
 
 const mapStateToProps = (state, props) => ({
-  payrollUuid: props.match.params.payrollUuid,
+  payrollUuid: props.match.params.payroll_uuid,
```

Comment:
I don't think this assignment should be part of map state to props as it's using props to map data to props. It can be extracted in component.

Reply:
In that case we can keep it, however I still think that's not a proper solution. I'm not sure if I understand this piece of code correctly but from my understanding this component expects to receive match.params in props and then maps it's attribute to the payrollUuid. If that's the case - my guess is that it should reduce complexity of getting the payrollUuid but in fact when it's still tightly coupled and if someone would like to reuse this component without passing match.params it'd fail. 

It feels like passing props.match.params.payroll_uuid as payrollUuid in parent component would be more object oriented.

Reply:
ok, Let's keep this and create improvement when we have time. This must be changed in many places in openIMIS because all it happenes in all pages when there is '/{uuid_object}' such pattern in URL for example:
- https://github.com/openimis/openimis-fe-contribution_plan_js/blob/31931ed7e61a4b2cecefd85a0f3f37fac4def3e6/src/pages/ContributionPlanPage.js#L65
- https://github.com/openimis/openimis-fe-policy_js/blob/69b8279edbe31c164af5a47ab6ffad9dc8cdd8ba/src/pages/PolicyPage.js#L83
- https://github.com/openimis/openimis-fe-contribution_js/blob/d64c6c217dfe92387e239f764bbe85ffecac984e/src/pages/ContributionPage.js#L67
- https://github.com/openimis/openimis-fe-invoice_js/blob/f84c479884ae3728db23549584749430152a9d2f/src/pages/InvoicePage.js#L121
- https://github.com/openimis/openimis-fe-medical_js/blob/ca219aa960aa1d6d8b3f835a3384c232614f4d2b/src/pages/MedicalItemOverviewPage.js#L30
I don't think that Page component could be reusable in another modules. If it is reusable it works in such way that there is a link to specific route where this page is registered. This is how Bill works in Payroll for example - we can use Searcher and go to Page with details of Bills without defining component.